src/displayapp/screens/settings/SettingWakeUp.cpp

```diff
@@ -90,18 +90,48 @@ bool SettingWakeUp::Refresh() {
 
 void SettingWakeUp::UpdateSelected(lv_obj_t *object, lv_event_t event) {
   if(event == LV_EVENT_VALUE_CHANGED) {
```

Comment:
Tested out your branch, and had the same problems as @Caton101. Tried to find the cause and I think at least one problem is using `LV_EVENT_VALUE_CHANGED`. Since this causes the `UpdateSelected` function to run when a value is changed, not just when it is clicked but also when changed with `lv_checkbox_set_checked`. So you get a loop of `UpdateSelected` functions when `lv_checkbox_set_checked` inside `UpdateSelected`. I'm not 100% sure but thats how i understand lvgl.

You could could use the `LV_EVENT_PRESSED` event instead but then you need to handle the conflict with lvgl's "hidden" checking/unchecking of the box. If not, then lvgl will try to check/uncheck and at the same time you check/uncheck inside `Updateselected`

Reply:
> handle the conflict with lvgl's "hidden" checking/unchecking of the box.  
Could you elaborate a bit about this? I'm not sure that I got what you mean.

Reply:
Ok I will try to explain it better.

We have our `event_handler`:
```
  static void event_handler(lv_obj_t * obj, lv_event_t event) {
    SettingWakeUp* screen = static_cast<SettingWakeUp *>(obj->user_data);
    screen->UpdateSelected(obj, event);
  }
```
which runs when a lvgl event occurs. Like when a button is clicked, dragged etc. More about events: [lvgl events](https://docs.lvgl.io/v7/en/html/overview/event.html#special-events). In our event_handler we call the function `screen->UpdateSelected(obj, event);` and then handles what to do with the event in this memberfunction of the screen.

`UpdateSelected` starts with this if statement: 
```  
void SettingWakeUp::UpdateSelected(lv_obj_t *object, lv_event_t event) {
    if(event == LV_EVENT_VALUE_CHANGED) {
```
```LV_EVENT_VALUE_CHANGED``` is a special event that is sent when the checkbox is checked/unchecked. [special event](https://docs.lvgl.io/v7/en/html/overview/event.html#special-events).

So when you click a button:

1. LVGL internally checks/unchecks the box and sends a LV_EVENT_VALUE_CHANGED event.
2. This event gets picked up by event_handler and passed to UpdateSelected
3. In UpdateSelected you change the values of this and other checkboxes multiple times with `lv_checkbox_set_checked`
4. If I'm correct each of your calls to lv_checkbox_set_checked sends out new LV_EVENT_VALUE_CHANGED events which in turn gets picked up by the event_handler and you get a loop.

Reply:
I see. Would introducing a "lock" variable, that would make UpdateSelected ignore events while it's set, and setting it when starting to handle a checkbox work?

Reply:
Yh that could work. Another solution could be to set the state of the checkboxes to disabled so lvgl dont do the checking/unchecking itself. And then instead use `LV_EVENT_CLICKED` and handle all the checking/unchecking in UpdateSelected